### PR TITLE
New package: py-google-common

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Review by alalazo — CHANGES_REQUESTED** [review]
What does this package do? It seems empty...

**Comment by iarspider** [comment]
It is a namespace package for other Google packages: https://packaging.python.org/guides/packaging-namespace-packages/

File: var/spack/repos/builtin/packages/py-google-common/package.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+class PyGoogleCommon(PythonPackage):
+    """Google namespace package"""
+
+    homepage = "http://pypi.python.org/pypi/google-common"
+    pypi     = "google-common/google-common-0.0.1.tar.gz"
+
+    version('0.0.1', sha256='add3cf494034fa11080e77cef2cc1f55a436a3353f5f5f95cc4f2b9788a678d1')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-google-common/package.py]: Can you check to see if this is used at run-time? The `src/google/__init__.py` file seems to import `pkg_resources`

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-google-common/package.py]: @adamjstewart fixed

**Comment by alalazo** [comment]
> It is a namespace package for other Google packages: https://packaging.python.org/guides/packaging-namespace-packages/

It seems to me that this one is empty and reading the link you posted every other subpackage must contain the same `__init__.py`. Can you point to some PyPI packages depending on this one?

---

To be more precise, [here](https://packaging.python.org/guides/packaging-namespace-packages/#pkg-resources-style-namespace-packages) it is reported that:

> Every distribution that uses the namespace package must include an identical __init__.py. If any distribution does not, it will cause the namespace logic to fail and the other sub-packages will not be importable. Any additional code in __init__.py will be inaccessible.

So I don't really understand the purpose of this empty package. Is there a reason you want to add it?

**Comment by iarspider** [comment]
In theory, this is needed for py-tensorflow and py-protobuf (and maybe py-grpcio), since they are all use `google` namespace. But maybe things are more complicated, I need to run more tests.

**Comment by alalazo** [comment]
> In theory, this is needed for py-tensorflow and py-protobuf (and maybe py-grpcio), since they are all use google namespace

I don't think it is reading the docs. I think though that each of these namespace packages will have the same snippet in their own `__init__.py` file. The reason I am picky is that this is a low traffic package that doesn't seem associated with Google at all and has not been updated since 2015. @adamjstewart any opinion on this?

**Comment by iarspider** [comment]
After reading the documentation again, it seems that this package is only needed for Python < 3.3: https://packaging.python.org/guides/packaging-namespace-packages/#pkg-resources-style-namespace-packages . Again, looks like I need to test this some more.

**Comment by adamjstewart** [comment]
Personally I'm fine with adding this package (it's very simple and unlikely to break) since we still support Python 2. I'll never use it, but someone might. I remember we had a lot of namespace issues with `py-basemap` and a few other packages once upon a time.

**Pull request closed without merging**